flags/src/main/java/com/yahoo/vespa/flags/FetchVector.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.flags;
+
+import javax.annotation.concurrent.Immutable;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * Denotes which RawFlag should be retrieved from {@link FlagSource} for a given {@link FlagId},
+ * as the raw flag may depend on the hostname, application, etc.
+ *
+ * @author hakonhall
+ */
+@Immutable
+public
+class FetchVector {
+    public enum Dimension {
+        /** Value from ZoneId::value */
+        ZONE_ID,
+        /** Value from ApplicationId::serializedForm */
+        APPLICATION_ID,
+        /** Fully qualified hostname */
+        HOSTNAME
+    }
+
+    private final Map<Dimension, String> map;
+
+    public FetchVector() {
+        this.map = Collections.emptyMap();
+    }
+
+    public static FetchVector fromMap(Map<Dimension, String> map) {
+        return new FetchVector(new HashMap<>(map));
```

Comment:
Use `EnumMap`?

Reply:
Unfortunately the EnumMap has a counterintuitive constructor taking a Map: If no element is found in the map, an IllegalArgumentException is thrown (because it's unable to get the Class without any instances), and this is detected at run-time which is uncomfortable. Just seems simpler to use Map.